docs/source/developers/guide/step_by_step/testing.rst

```diff
@@ -62,29 +62,81 @@ In this section we outline steps needed for unit testing in Arrow.
 
       If the tests start failing, try to recompile
       PyArrow or C++.
-      
+
       .. note::
 
          **Recompiling Cython**
 
          If you only make changes to `.py` files, you do not need to
          recompile PyArrow. However, you should recompile it if you make
          changes in `.pyx` or `.pxd` files.
-        
+
          To do that run this command again:
 
          .. code:: console
 
             $ python setup.py build_ext --inplace
 
       .. note::
-		
+
          **Recompiling C++**
 
          Similarly, you will need to recompile the C++ code if you have
          made changes to any C++ files. In this case,
-         re-run the cmake commands again. 
+         re-run the cmake commands again.
 
    .. tab:: R tests
 
-      .. TODO
+      We use `testthat <https://testthat.r-lib.org/index.html>`_ for unit testing in R. More specifically, we use the `3rd edition of testthat <https://testthat.r-lib.org/articles/third-edition.html>`_. On rare occasions we might want the behaviour of the 2nd edition of testthat, which is indicated by ``testthat::local_edition(2)``.
+
+      **Structure**
+
+      Expect the usual testthat folder structure:
+      .. TODO make sure the format the dir tree below is formatted nicely (maybe as a code block)
+      tests
+        ├── testthat      # unit test files live here
+        └── testthat.R    # runs tests when R CMD check runs (e.g. with `devtools::check()`)
+
+      Usually, most files in the ``R/`` sub-folder have a corresponding test file in ``tests/testthat``.
+
+      **Running tests**
+
+      To run all tests in a package locally use ``devtools::test()`` in the R console. Alternatively, you can use ``make test`` in the shell.
```

Comment:
```suggestion
      To run all tests in a package locally use this command in the R console:

      .. code-block:: R

         devtools::test()

      Alternatively, you can use this in the shell:

      .. code:: console

         $ make test
```

It is easier to find the necessary info in the html like so, with code being highlighted.